### PR TITLE
add more ce impl for sft

### PR DESCRIPTION
**Pull request by samsja**
Adding ce sft implemention for reducing memory usage:

* Lieger implementation
* default torch implementation

---

> [!NOTE]
> Adds a `loss_impl` option to choose torch or Liger cross-entropy and uses the selected implementation during SFT training.
> 
> - **Config (`src/prime_rl/trainer/sft/config.py`)**:
>   - Add `SFTTrainerConfig.loss_impl` (`"liger"|"torch"`, default `"torch"`) to select CE loss implementation.
> - **Training (`src/prime_rl/trainer/sft/train.py`)**:
>   - Import and instantiate `LigerCrossEntropyLoss` or `CrossEntropyLoss` based on `config.loss_impl`.
>   - Replace use of `torch.nn.functional.cross_entropy` with the selected `ce_loss` in loss computation.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 859ea1bf629da14887285f6300c05fd016791b0f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/prime_rl/trainer/sft/train.py [hunk 1]
```diff
@@ -130,6 +132,14 @@ def train(config: SFTTrainerConfig):
         f"Starting from step {progress.step} (total_tokens={progress.total_tokens}, total_samples={progress.total_samples}, dataset_state={dataloader.state_dict()['dataset_state']})"
     )
 
+    match config.loss_impl:
+        case "liger":
+            ce_loss = LigerCrossEntropyLoss(reduction="none")
+        case "torch":
+            ce_loss = partial(cross_entropy, reduction="none")
```
Comment (mikasenghaas) [inline src/prime_rl/trainer/sft/train.py]: class here for better vibes

Comment (samsja) [reply src/prime_rl/trainer/sft/train.py]: done

**Review by mikasenghaas — APPROVED** [review]
